src/pages/display.tsx

```diff
@@ -127,7 +136,10 @@ class Display extends Component<DisplayProps, DisplayComponentState> {
       case views.DISPLAY_WAITING_TO_START:
         return <WaitingToStart />
       case views.DISPLAY_GAME:
-        return <Game socket={socket!} onGameOver={this.gameOver} />
+        if (!socket) {
+          throw new Error("Socket is not available")
```

Comment:
What does this render if we don't have the socket here?

Reply:
it shouldn't occur ever but I don't like to use the `!` operator in TS as it makes the code unreliable. We can add an error boundary and render a basic "uh oh something happened" page if any error is thrown.

Reply:
Sounds better. It looks a bit hazard to throw in a switch without a default case or catch. Looks like we don't know what to do if shit hits the fan.

Reply:
Added an error page and if the socket is falsy we'll render the "Something went wrong"

Reply:
👍